apps/hash-graph/postgres_migrations/V5__ontology_metadata_functions.sql

```diff
@@ -39,6 +45,20 @@ DECLARE
 BEGIN
   _ontology_id := gen_random_uuid();
 
+  BEGIN
+    INSERT INTO base_uris (
+      "base_uri"
+    ) VALUES (
+      create_external_ontology_id.base_uri
+    );
+  EXCEPTION WHEN unique_violation THEN
+    IF EXISTS (SELECT FROM ontology_ids NATURAL JOIN ontology_owned_metadata WHERE ontology_ids.base_uri = create_external_ontology_id.base_uri) THEN
+      RAISE EXCEPTION 'Owned ontology with base_uri `%` already exists',
+        create_external_ontology_id.base_uri
+      USING ERRCODE = 'invalid_parameter_value';
+    END IF;
+  END;
```

Comment:
Is this nested transaction required for us to catch the unique violation?

And for my understanding, is this explictitly ignoring the exception if the base URI already exists for a non-owned ontology type?

Reply:
I was not able to come up with an easier solution 🙁 

Yes, this was the only way I found to catch the exception. Happy to change this if you have a better idea 😄

Reply:
I see, I don't have a better solution, just wanted to know the context. This makes sense, though :)